src/java/org/apache/cassandra/service/ClientState.java

```diff
@@ -100,8 +100,8 @@
             }
             catch (Exception e)
             {
-                JVMStabilityInspector.inspectThrowable(e);
-                logger.info("Cannot use class {} as query handler ({}), ignoring by defaulting on normal query handling", customHandlerClass, e.getMessage());
+                logger.error("Cannot use class {} as query handler ({})", customHandlerClass, e.getMessage());
+                JVMStabilityInspector.killCurrentJVM(e, false);
```

Comment:
Since the contextualized exception is already being logged above and the root cause is well known, I think quiet parameter could be **true**, otherwise it will print the stackTrace to System.err and to the logger.error "JVM state determined to be unstable.  Exiting forcefully due to:" which is not really necessary in this case.

Overall, this is how Im currently handling this situation in my custom query handler:
```
MyQueryHandler() {
   try {
     configure();
   } catch (ConfigurationException e) {
     log.error("Error registering MyQueryHandler", e);
     JVMStabilityInspector.killCurrentJVM(e, true);
   }
}
```

Reply:
done
this is the output I get:
```
ERROR [main] 2021-07-01 16:16:33,552 ClientState.java:103 - Cannot use class org.apache.cassandra.cql3.CustomPayloadMirroringQueryHandler2 as query handler
org.apache.cassandra.exceptions.ConfigurationException: Unable to find QueryHandler class 'org.apache.cassandra.cql3.CustomPayloadMirroringQueryHandler2'
	at org.apache.cassandra.utils.FBUtilities.classForName(FBUtilities.java:720)
	at org.apache.cassandra.utils.FBUtilities.construct(FBUtilities.java:753)
	at org.apache.cassandra.service.ClientState.<clinit>(ClientState.java:98)
	at org.apache.cassandra.cql3.QueryProcessor$InternalStateInstance.<init>(QueryProcessor.java:129)
	at org.apache.cassandra.cql3.QueryProcessor$InternalStateInstance.<clinit>(QueryProcessor.java:123)
	at org.apache.cassandra.cql3.QueryProcessor.internalQueryState(QueryProcessor.java:174)
	at org.apache.cassandra.cql3.QueryProcessor.prepareInternal(QueryProcessor.java:312)
	at org.apache.cassandra.cql3.QueryProcessor.executeInternal(QueryProcessor.java:322)
	at org.apache.cassandra.db.SystemKeyspace.checkHealth(SystemKeyspace.java:973)
	at org.apache.cassandra.service.StartupChecks$10.execute(StartupChecks.java:442)
	at org.apache.cassandra.service.StartupChecks.verify(StartupChecks.java:132)
	at org.apache.cassandra.service.CassandraDaemon.runStartupChecks(CassandraDaemon.java:487)
	at org.apache.cassandra.service.CassandraDaemon.setup(CassandraDaemon.java:262)
	at org.apache.cassandra.service.CassandraDaemon.activate(CassandraDaemon.java:763)
	at org.apache.cassandra.service.CassandraDaemon.main(CassandraDaemon.java:887)
Caused by: java.lang.ClassNotFoundException: org.apache.cassandra.cql3.CustomPayloadMirroringQueryHandler2
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at org.apache.cassandra.utils.FBUtilities.classForName(FBUtilities.java:716)
	... 14 common frames omitted

Process finished with exit code 100
```